contracts/FundManagement.sol

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity 0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * Contract to allow Tracer DAO to delegate funds to managers to spend as they see fit. 
+ * Managers can request to withdraw funds assigned to them. The DAO may take back the funds at any time.
+ */
+contract FundManagement is Ownable {
+    struct Fund {
+        uint256 totalAmount; // Total amount of tokens assigned to the manager
+        address asset;
+        uint256 requestedWithdrawTime; // Timestamp which the manager may withdraw the pendingWithdrawAmount
+        uint256 pendingWithdrawAmount; // Total amount of tokens the manager is pending to withdraw
+    }
+
+    using SafeERC20 for IERC20;
+
+    uint256 public requestWindow = 2 days;
+    mapping(address => mapping(uint256 => Fund)) public funds;
+    mapping(address => uint256) public numberOfFunds;
+    mapping(address => uint256) public locked;
+
+    event CreateFund(address indexed manager, address indexed asset, uint256 amount);
+    event RequestFunds(address indexed to, uint256 amount);
+    event Claim(address indexed to, uint256 amount);
+    event Clawback(address indexed account, uint256 fundNumber);
+
+    constructor() {}
+
+    function requestFunds(uint256 fundNumber, uint256 amount) external {
```

Comment:
I think `fundNumber` should be bounds checked as it's an offset into an array. Regardless of Solidity's behaviour, this would aid debugging.

Reply:
Its an index into a mapping, so in this case will return a zero value for all fields.

However you raise a good point, you could request a withdraw of `amount=0` and this transaction would work and simply increment the `requestWithdrawTime` which is not ideal.